backend-transit/src/main/java/eu/socialedge/hermes/backend/transit/domain/Route.java

```diff
@@ -64,7 +64,7 @@ public Route(String code, VehicleType vehicleType, List<Station> stations, Shape
         this.vehicleType = notNull(vehicleType);
         this.stations = new ArrayList<>(notEmpty(stations));
 
-        if (!containsAllStations(shape))
+        if (!containsAllStations(notNull(shape)))
```

Comment:
Consider creating new constructor ```Route(String code, VehicleType vehicleType, List<Station> stations)``` (without Shape). Seems it can be **not** ```@NotNull``` since you removed this annotation and Shape became optional.